src/components/course/CoursePage.jsx

```diff
@@ -159,38 +149,74 @@ const CoursePage = () => {
         availableCourseRuns: getAvailableCourseRuns(courseDetails),
         userEnrollments,
         userEntitlements,
-        userSubsidyApplicableToCourse,
-        legacyUserSubsidyApplicableToCourse,
         catalog,
         courseReviews,
         algoliaSearchParams,
         courseRecommendations: {
           allRecommendations: allRecommendations?.slice(0, 3),
           samePartnerRecommendations: samePartnerRecommendations?.slice(0, 3),
         },
-        isPolicyRedemptionEnabled,
-        redeemabilityPerContentKey,
       };
     },
     [
-      userSubsidyApplicableToCourse,
-      legacyUserSubsidyApplicableToCourse,
-      isLoadingCourseData,
-      isLoadingAccessPolicyRedemptionStatus,
+      isLoadingAny,
       courseData,
       courseRecommendations,
       courseReviews,
       algoliaSearchParams,
-      isPolicyRedemptionEnabled,
-      redeemabilityPerContentKey,
     ],
   );
 
+  const {
+    userSubsidyApplicableToCourse,
+    missingUserSubsidyReason,
+  } = useUserSubsidyApplicableToCourse({
+    courseData,
+    redeemableSubsidyAccessPolicy,
+    isPolicyRedemptionEnabled,
+    subscriptionLicense,
+    courseService,
+    couponCodes,
+    canEnrollWithEnterpriseOffers,
+    enterpriseOffers,
+    onSubscriptionLicenseForCourseValidationError,
+    missingSubsidyAccessPolicyReason,
+    enterpriseAdminUsers,
+    courseListPrice: (
+      courseState?.activeCourseRun?.firstEnrollablePaidSeatPrice
+      || getEntitlementPrice(courseState?.course?.entitlements)
+    ),
+  });
+
+  const [coursePrice, currency] = useCoursePriceForUserSubsidy({
+    courseEntitlements: courseState?.course?.entitlements,
+    activeCourseRun: courseState?.activeCourseRun,
+    userSubsidyApplicableToCourse,
+  });
+
+  useEffect(() => {
+    // Redirect if path does not contain course type
```

Comment:
Moved into a `useEffect` from below, so `history.replace` doesn't run on every re-render on of `CoursePage` when the below conditions are met; instead, it only gets called when the dependent props are changed.